### PR TITLE
chore: cleanup unused and deprecated code

### PR DESCRIPTION
**Pull request by gregdhill**
Making this PR to run the benchmarks for the release, no functional changes are included.

### CODE REVIEW

File: parachain/src/service.rs [hunk 1]
```diff
@@ -254,11 +254,20 @@ where
...
-        config.default_heap_pages,
-        config.max_runtime_instances,
-        config.runtime_cache_size,
+    let heap_pages = config
+        .default_heap_pages
+        .map_or(DEFAULT_HEAP_ALLOC_STRATEGY, |h| HeapAllocStrategy::Static {
+            extra_pages: h as _,
+        });
+
+    let executor = NativeElseWasmExecutor::<Executor>::new_with_wasm_executor(
```
[the remark this answers is not in the document]
Comment (sander2) [reply parachain/src/service.rs:263]: Do you have a source on this? The native executor is a lot more efficient so isn't it preferable?

[the remark this answers is not in the document]
Comment (gregdhill) [reply parachain/src/service.rs:263]: Originally I saw someone commented on the PR which deprecates the `new` constructor [here](https://github.com/paritytech/substrate/pull/13740) but then I found [this](https://github.com/paritytech/substrate/issues/7288) issue with a more detailed explanation. Statemint has also already switched to wasm runtime only, see [this PR](https://github.com/paritytech/cumulus/pull/1054).

**Review by sander2 — APPROVED** [review]

**Pull request merged**